### PR TITLE
Move region into secret of openstack

### PR DESCRIPTION
**Pull request by jichenjc**
Openstack Region should be put into secret..

**What this PR does / why we need it**:
'Region' is secret in openrc file (params to link to keystone of openstack)
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```
Region should be a secret , not a common user settings
```

### CODE REVIEW

File: pkg/cloudprovider/provider/openstack/provider.go [hunk 1]
```diff
@@ -120,6 +120,12 @@ func (p *provider) getConfig(s v1alpha1.ProviderConfig) (*Config, *providerconfi
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("failed to get the value of \"password\" field, error = %v", err)
 	}
+	// Ignore Region not found as Region might not be found and we can default it later
+	c.Region, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.Region, "OS_REGION_NAME")
+        if err != nil {
+		glog.V(0).Infof("Region from configuration or environment variable not found")
```
Comment (alvaroaleman) [inline pkg/cloudprovider/provider/openstack/provider.go]: Can you please make this a `glog.V(6).Infof` as its debug log? Other than that, pull LGTM

Comment (jichenjc) [reply pkg/cloudprovider/provider/openstack/provider.go]: ok, will do

**Pull request merged**